x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/InferTrainedModelDeploymentAction.java

```diff
@@ -39,17 +41,25 @@ public InferTrainedModelDeploymentAction() {
 
     public static class Request extends BaseTasksRequest<Request> implements ToXContentObject {
 
-        public static final String DEPLOYMENT_ID = "deployment_id";
+        public static final ParseField DEPLOYMENT_ID = new ParseField("deployment_id");
         public static final ParseField INPUT = new ParseField("input");
+        public static final ParseField TIMEOUT = new ParseField("timeout");
 
-        private static final ObjectParser<Request, Void> PARSER = new ObjectParser<>(NAME, Request::new);
+        private static final TimeValue DEFAULT_TIMEOUT = TimeValue.timeValueSeconds(10);
```

Comment:
The spec says this should be 20 seconds.

Reply:
The spec update is for [StartTrainedModelDeploymentAction](https://github.com/elastic/elasticsearch/blob/63012c8a40742829806b44cb6577b05d9ab25a83/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/StartTrainedModelDeploymentAction.java#L37)

For StartTrainedModelDeploymentAction `timeout` is a query parameter which is why it is included in the spec. Here, in infer `timeout` is in the request body content and not documented in the spec. Hence my question about about timeout being a query parameter for the infer action rather than in the body. 

We are not consistent with timeout parameters [Close Job](https://github.com/elastic/elasticsearch/blob/68817d7ca29c264b3ea3f766737d81e2ebb4028c/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/job/RestCloseJobAction.java#L45) takes timeout _only_ in the body whereas [Open Job](https://github.com/elastic/elasticsearch/blob/68817d7ca29c264b3ea3f766737d81e2ebb4028c/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/job/RestOpenJobAction.java#L49) accepts it either in the body or URL.

I am now leaning towards putting the infer `timeout` option in the URL as a query parameter. It is cleaner and to do otherwise adds complexity to any code that is building the input request doc.


Returning to your comment, documenting the default in the spec just creates a maintenance burden